pgcontents/crypto.py

```diff
@@ -221,8 +221,26 @@ def single_password_crypto_factory(password):
     The factory here returns a ``FernetEncryption`` that uses a key derived
     from ``password`` and salted with the supplied user_id.
     """
+    @MemoizedCryptoFactory
     def factory(user_id):
         return FernetEncryption(
             Fernet(derive_single_fernet_key(password, user_id))
         )
     return factory
+
+
+class MemoizedCryptoFactory(object):
```

Comment:
The name for this feels a little misleading; nothing about this knows or cares about the fact that it's memoizing encryption stuff.  My inclination here would be to either pull the `FernetEncryption` construction logic into this class, or to just convert this into a classic `@memoize` decorator:

```python
from functools import wraps

def memoize_single_arg(f):
    memo = {}
    @wraps(f)
    def memoized_f(arg):
        try:
            return memo[arg]
        except KeyError:
            result = memo[arg] = f(arg)
            return result
    return memoized_f
```